### DISCUSSION TITLE
Special handling required for names with spaces, e.g. "St John"

### DISCUSSION THREAD

**Discussion by hartenthaler** [fisharebest/gedcom-name#18] [General]
Are there names with a space in them?

For me "St John" is not a good example. For me, "St" or "Saint" is a title like a noble title "King". I would code it

> 1 TITL St
> 2 NAME John

John was born as John. Years after he died, the church gave him the title "Saint".

**Comment by Norwegian-Sardines** [comment]
For an example, Bill St. John is a Wine Writer for several newspapers.

1 NAME Bill /St. John/

**Comment by hartenthaler** [comment]
I see! so we can add rule #41 to the "Falsehoods Programmers Believe About Names". There is nothing with names that is impossible.

**Comment by hartenthaler** [comment]
And his surname is flexible. Sometimes he is written "St. John", sometimes "St John". So his name contains space and dot. At least the dot is optional. Strange.

But could be there a problem? As far as I understood the GEDCOM 7 standard, names can contain nearly all characters: only "/" and tab are not allowed.

**Comment by fisharebest** [comment]
`St John` is an English given name.  It is pronounced "Sinjun".

I am sure that other languages will have examples.

**Comment by Norwegian-Sardines** [comment]
> Are there names with a space in them?

Jill St. John an actress.
The Van Halen brothers (Eddie, Alex and Wolfgang) formed a rock band.

**Comment by hartenthaler** [comment]
Names can contain many different types of characters, such as blanks, dots, umlauts, diacritic, Chinese, ...

But they should not contain tabs, commas, semicolons, slashes, backslashes, ...

\* and " have a special meaning and should not be part of the name itself.

**Comment by Tim-Partridge** [comment]
Spanish people can have two family names with a space between them. Family names change every generation. A child has the father's first family name followed by the mother's first family name. (Source of information my Spanish teacher.)

English people can have two names combined with a hyphen. These were often the result of marriages between noble families where the bride's side insisted on the groom changing his surname to include the bride's surname as well. In more recent times couples may wish to show equality of the partners by both changing their family name to a new name that has both thier old ones.

**Reply by Norwegian-Sardines** [reply]
Much of the Spanish, Portuguese and other multi surname discussion can be found in this thread:
https://github.com/fisharebest/gedcom-name/discussions/8
